### DISCUSSION TITLE
Actual native traffic analytics

### DISCUSSION THREAD

**Discussion by port19x** [community/community#48900] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

It would be nice to have actual traffic analytics for github pages.
Like native, server side, non-js, dead simple analytics.

I'm not talking about repo traffic analytics, those have existed for ages, that is not the problem and it's not useful for this usecase. I'm aware of those.

Google analytics, matomo, goat counter, etc. don't cut it either. I don't want to have javascript executing on my visitors browser. On top of that, this couldn't track something like an rss feed, which is a common mechanism for subscribing to smaller blogs, so repeated visitors that use a feed reader wouldn't be accounted for.

[netlify analytics](https://www.netlify.com/products/analytics/) is one offering of a competing static site host, but costs a staggering 9$ per month per site.
Another option that I'm seriously looking into is going the full manual route: setting up a VPS with digital ocean via ansible, then running ngingx on there and checking access logs every once in a while.

This should be a very easy feature to implement, as whatever http server actually serves the content under the hood probably already has some form of logging.
Furthermore I wouldn't even care about detailed analytics, like platform or even country, google analytics exists for freaks who want to know all the details about their visitors.
Just tell me in very simple terms how many requests I get on each day, and if you're extra generous maybe even split that up by page.

**Comment by dellagustin-sap** [comment]
Seems like a duplicate of https://github.com/orgs/community/discussions/31474, but I would love to have this feature.

**Comment by Venkatesh-KCET** [comment]
@dellagustin-sap is correct its a duplicate of https://github.com/orgs/community/discussions/31474

This feature is available in GitHub.

You can follow the steps below:
- Go to your code repository
- Look for the 'Insights' tab, click on it
- Then click on 'Traffic' tab
- Once there, you will see a variety of metrics like visitors, how many people cloned your repo etc

**Reply by dellagustin-sap** [reply]
@Venkatesh-KCET , the feature you described is for the repo, but not for GitHub Pages.
As stated by @port19x:
> It would be nice to have actual traffic analytics for **github pages**.
> Like native, server side, non-js, dead simple analytics.
>
> I'm not talking about repo traffic analytics, those have existed for ages, that is not the problem and it's not useful for this usecase. I'm aware of those.

**Reply by Venkatesh-KCET** [reply]
GitHub has provided native "[GitHub Traffic Analytics](https://github.com/blog/1672-introducing-github-traffic-analytics)" since 2014.

Besides tracking visitors to your repository, it provides a 'content' panel where you can see the traffic to individual pages within that repository.

1.  Navigate to your desired repository
2.  Go to 'Insights' and select 'Traffic'.

This is what it looks like:

[![enter image description here](https://i.stack.imgur.com/aNo9Q.png)](https://i.stack.imgur.com/aNo9Q.png)

If your GitHub page doesn't have enough traffic, or it's new, you'll see this:

[![enter image description here](https://i.stack.imgur.com/cGpvu.jpg)](https://i.stack.imgur.com/cGpvu.jpg)

**Reply by dellagustin-sap** [reply, marked as answer]
@Venkatesh-KCET I believe that in your last comment you are referring to https://webapps.stackexchange.com/a/113028.
I think that _you can see the traffic to individual pages within that repository_ means subpages of the repository, like pull requests, issues, specific files, etc... but not to content served as [GitHub Pages](https://pages.github.com/) - [About GitHub Pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages).
The official documentation for this feature ([Viewing traffic to a repository](https://docs.github.com/en/repositories/viewing-activity-and-data-for-your-repository/viewing-traffic-to-a-repository)) does not mention anything about GitHub Pages.
